distribution/push_v2_test.go

```diff
@@ -637,7 +638,15 @@ type mockRepo struct {
 
 var _ distribution.Repository = &mockRepo{}
 
-func (m *mockRepo) Named() reference.Named {
+// Named() must return a distref.Named here, not containerd's Named interface;
+// while both interfaces are identical, Golang considers it an incompatible
+// type otherwise:
+//
+//     distribution/push_v2_test.go:404:4: cannot use repo (type *mockRepo) as type distribution.Repository in field value:
+//     *mockRepo does not implement distribution.Repository (wrong type for Named method)
+//     have Named() docker.Named
+//     want Named() "github.com/docker/docker/vendor/github.com/docker/distribution/reference".Named
+func (m *mockRepo) Named() distref.Named {
```

Comment:
As it looks like the "interface" issues really only affects this Mock, I decided to import the "old" package here. "production" code doesn't need the old interface

Not sure if there's a better solution for this; perhaps we should replace the `distribution.Repository` interface with something local if possible (haven't dug too deeply into that yet)